source/mbed_trace.c

```diff
@@ -28,7 +28,7 @@
 #include "mbed-trace/mbed_trace_ip6tos.h"
 #endif
 
-#if defined(_WIN32) || defined(__unix__) || defined(__unix) || defined(unix) || defined(YOTTA_CFG)
+#if defined(_WIN32) || defined(__unix__) || defined(__unix) || defined(unix) || (defined(YOTTA_CFG) && !defined(NDEBUG))
```

Comment:
`NDEBUG` should be avoided, because it's very nanostack specific. I think this could be done different way